rskj-core/src/main/java/co/rsk/net/sync/DecidingSyncState.java

```diff
@@ -91,4 +127,12 @@ private Optional<Long> getPeerBestBlockNumber(Peer peer) {
         return Optional.ofNullable(peersInformation.getPeer(peer))
                 .flatMap(pi -> Optional.ofNullable(pi.getStatus()).map(Status::getBestBlockNumber));
     }
+
+    private boolean checkGenesisConnected(BlockStore blockStore) {
```

Comment:
wondering if we could just check if `blockStore.getMinNumber() == 0` here?

Reply:
yeah, actually that was my first attempt, but then I found the `org.ethereum.core.Block#isGenesis` method and thought that it was safer to use it

but if using `== 0` is safe I'll go for it, as it will be faster (we avoid calling `blockStore.getChainBlockByNumber`)

thanks Vlad 👍